File: agentic-crawler/src/schemas/faq.ts
import { z } from 'zod';

export const FaqPairSchema = z.object({
  question: z.string().min(1),
  answer: z.string().min(1),
  category: z.string().optional(),
  url: z.string().optional(),
});

export type FaqPair = z.infer<typeof FaqPairSchema>;

export const FaqOutputSchema = z.object({
  domain: z.string(),
  startUrl: z.string(),
  scrapedAt: z.string(),
  totalPairs: z.number(),
  pairs: z.array(FaqPairSchema),
});

export type FaqOutput = z.infer<typeof FaqOutputSchema>;

File: agentic-crawler-py/src/schemas/faq.py
"""Pydantic schemas for FAQ extraction."""

from __future__ import annotations

from pydantic import BaseModel


class FaqPair(BaseModel):
    question: str
    answer: str
    category: str | None = None
    url: str | None = None


class FaqOutput(BaseModel):
    domain: str
    start_url: str
    scraped_at: str
    total_pairs: int
    pairs: list[FaqPair]
